beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/AbstractBlockFactoryTest.java

```diff
@@ -373,27 +197,39 @@ private BeaconBlock assertBlockCreated(
     assertThat(block.getBody().getAttesterSlashings()).isEqualTo(attesterSlashings);
     assertThat(block.getBody().getProposerSlashings()).isEqualTo(proposerSlashings);
     assertThat(block.getBody().getVoluntaryExits()).isEqualTo(voluntaryExits);
-    assertThat(block.getBody().getGraffiti()).isEqualTo(graffiti);
+    assertThat(block.getBody().getGraffiti()).isNotNull();
 
-    if (spec.getGenesisSpec().getMilestone().isGreaterThanOrEqualTo(SpecMilestone.CAPELLA)) {
+    if (milestone.isGreaterThanOrEqualTo(SpecMilestone.CAPELLA)) {
       assertThat(block.getBody().getOptionalBlsToExecutionChanges())
           .isPresent()
           .hasValue(blsToExecutionChanges);
     } else {
       assertThat(block.getBody().getOptionalBlsToExecutionChanges()).isEmpty();
     }
 
-    return block;
+    if (milestone.isGreaterThanOrEqualTo(SpecMilestone.DENEB)) {
```

Comment:
I worry on broad usage of such complex milestone based assertions in big methods. There is a chance we could assert not the thing we expect in tests. For example here, input block and milestone variable could be from different milestones etc. 
Just a feeling

Reply:
But the milestone is based on the new block created, so will always align. Changed the test a little bit to mimic the blsExecutionChanges for Capella